app/Http/Controllers/Geolocation/GeolocationController.php

```diff
@@ -32,10 +32,22 @@ public function getGeolocationByUserQuery(Request $request)
         ]);
 
         $search = $this->nominatim->newSearch();
-
+        
         $search->query($request->input('postcode'));
 
         $result = $this->nominatim->find($search);
+        
+        if(!$result || !$result[0]){
+            
+            $this->validate($request, [
+                'city' => 'required',
+            ]);
```

Comment:
By doing this, the request will return an error, if the city is not set. We should not do this, as the only required argument for this endpoint is the postcode, and there are some request made where the CITY is not actually available.

If you just change this to fail gracefully when city is not present ) add it on the if statement, i can then merge this in

Reply:
Yeah...I'll do that.